### PR TITLE
Peer syncing toggle

### LINKED ISSUE #4606: Basic peer syncing: add an ability to enable/disable it in settings
The work regarding adding basic peersyncing is done by @cammellos in this PR: https://github.com/status-im/status-go/pull/4543
This issue is about allowing to enable it through settings. That includes:
- adding a field in the settings database 
- using that field to enable/disable
- add ui code in settings
- add enpdoint

The next step would be to do that on client side as well.

### PR DESCRIPTION
**Pull request by alwx**
Fixes https://github.com/status-im/status-go/issues/4606

- adds field to the database
- uses that field to enable/disable peer syncing
- adds endpoint to enable/disable
- syncs these changes between clients

Comes with https://github.com/status-im/status-mobile/pull/18906 (a client-side code that adds toggle)

### CODE REVIEW

File: multiaccounts/settings/columns.go [hunk 1]
```diff
@@ -506,28 +506,44 @@ var (
 		dBColumnName:   "mnemonic_was_not_shown",
 		valueHandler:   BoolHandler,
 	}
+	PeerSyncingEnabled = SettingField{
+		reactFieldName: "peer-syncing-enabled?",
+		dBColumnName:   "peer_syncing_enabled",
+		valueHandler:   BoolHandler,
+		syncProtobufFactory: &SyncProtobufFactory{
```
Comment (cammellos) [inline multiaccounts/settings/columns.go]: We don't need to sync this across installations

File: multiaccounts/settings/sync_protobuf_factories.go [hunk 2]
```diff
@@ -341,6 +341,33 @@ func stickersRecentStickersProtobufFactoryStruct(s Settings, clock uint64, chatI
 	return buildRawStickersRecentStickersSyncMessage(srs, clock, chatID)
 }
 
+// PeerSyncing
+
+func buildRawPeerSyncingEnabledSyncMessage(v bool, clock uint64, chatID string) (*common.RawMessage, *protobuf.SyncSetting, error) {
```
Comment (cammellos) [inline multiaccounts/settings/sync_protobuf_factories.go]: this can be removed as no need to sync it

File: protocol/messenger_mailserver.go [hunk 3]
```diff
@@ -1019,6 +1019,14 @@ func (m *Messenger) ToggleUseMailservers(value bool) error {
 	return nil
 }
 
+func (m *Messenger) TogglePeerSyncing(value bool) error {
```
Comment (cammellos) [inline protocol/messenger_mailserver.go]: maybe best to put it in `protocol/messenger_settings.go`
and use a `request` object

similarly to:

```
func (m *Messenger) SetLightClient(request *requests.SetLightClient) error {
	return nodecfg.SetLightClient(m.database, request.Enabled)
}
```

**Review by cammellos — CHANGES_REQUESTED** [review]
Looks good for the most part, the only thing is that we don't need to sync it across devices, and some minor neatpicks on naming/parameters, great work!

**Comment by alwx** [comment]
@cammellos everything was fixed in the latest commit

File: protocol/protobuf/sync_settings.proto [hunk 4]
```diff
@@ -39,6 +39,7 @@ message SyncSetting {
     SHOW_COMMUNITY_ASSET_WHEN_SENDING_TOKENS = 19;
     DISPLAY_ASSETS_BELOW_BALANCE = 20;
     DISPLAY_ASSETS_BELOW_BALANCE_THRESHOLD = 21;
+    PEER_SYNCING_ENABLED = 22;
```
Comment (cammellos) [inline protocol/protobuf/sync_settings.proto]: no need for this anymore

File: services/ext/api.go [hunk 5]
```diff
@@ -1490,6 +1490,10 @@ func (api *PublicAPI) ToggleUseMailservers(value bool) error {
 	return api.service.messenger.ToggleUseMailservers(value)
 }
 
+func (api *PublicAPI) TogglePeerSyncing(value bool) error {
```
Comment (cammellos) [inline services/ext/api.go]: last thing, if you could use  `protocol/requests/toggle_peer_syncing.go`  for this,l similarly as we do for other

**Review by cammellos — APPROVED** [review]

**Pull request merged**